lib/mix/tasks/merge_projects.ex

```diff
@@ -89,30 +89,34 @@ defmodule Mix.Tasks.Lightning.MergeProjects do
     write_output(output, output_path)
   end
 
-  defp perform_merge(source_project, target_project) do
+  defp perform_merge(source_data, target_data) do
+    source_project = atomize_keys(source_data)
+    target_project = atomize_keys(target_data)
+
     MergeProjects.merge_project(source_project, target_project)
   rescue
-    e in KeyError ->
+    ArgumentError ->
       Mix.raise("""
-      Failed to merge projects - missing required field: #{inspect(e.key)}
-
-      #{Exception.message(e)}
+      Failed to merge projects - encountered unknown field in JSON
 
-      This may indicate incompatible or corrupted project state files.
-      Please verify both files are valid Lightning project exports.
+      This may indicate the JSON contains invalid or unexpected fields.
+      Please ensure both files are valid Lightning project exports.
       """)
+  end
 
-    e ->
-      Mix.raise("""
-      Failed to merge projects
-
-      #{Exception.message(e)}
+  defp atomize_keys(data) when is_map(data) do
+    Map.new(data, fn {key, value} ->
+      atom_key = if is_binary(key), do: String.to_existing_atom(key), else: key
+      {atom_key, atomize_keys(value)}
```

Comment:
@elias-ba If you are going to be atomizing things that are not keys - maybe this should be called just `atomize`, or `maybe_atomize`?